### PR TITLE
Accommodate template changes new batch ingest - Multi-file upload

### PR DESCRIPTION
**Pull request by lagoan**
## Context

The new batch ingest workflow requires the ability to specify multiple files per item row as it was possible in the previous command line batch ingest workflow.

Related to #2943, #2853

## What's New

Changes in the code include:
+ Accept multiple files on batch ingest job with separated with the '|' character
+ Validation for multiple files added through the interface split by the '|' character
+ Tests to check multiple files can be handled by the batch ingestion

### CODE REVIEW

**Review by pgwillia — CHANGES_REQUESTED** [review]
Looking good!  I checked the recordings for secrets and don't see any.  Thank you for the code comments explaining the method for creating the cassettes.  

My main concern is around error handling and edge conditions.  I didn't see a test for validation failure.  Once we think that through I think this is ready to go!

File: test/jobs/batch_ingestion_job_test.rb [hunk 1]
```diff
@@ -66,4 +66,63 @@ def batch_ingest.processing!
...
+    # cassette.response.body.force_encoding('UTF-8')
+
+    # VCR.use_cassette('google_fetch_spreadsheet_multiple_files_in_one_row') do
+    #   uri =  URI("https://sheets.googleapis.com/v4/spreadsheets/RANDOMSPREADSHEETID/values/'Data'!A:X?key=test-google-developer-key")
+    #   Net::HTTP.get(uri)
+    # end
+
+    assert_difference('Item.count', +3) do
+      assert_difference('ActiveStorage::Blob.count', +9) do
+        VCR.use_cassette('google_fetch_spreadsheet_multiple_files_in_one_row',
```
[the remark this answers is not in the document]
Comment (lagoan) [reply test/jobs/batch_ingestion_job_test.rb:97]: I will take a look into this, it would definitely make for cleaner code.

File: app/jobs/batch_ingestion_job.rb [hunk 2]
```diff
@@ -121,9 +124,15 @@ def item_ingest(batch_ingest, item_data)
...
-  def file_ingest(item, item_file, google_credentials)
-    file = google_credentials.download_file(item_file.google_file_id, item_file.google_file_name)
-    item.add_and_ingest_files([file])
+  def files_ingest(item, item_files, google_credentials)
+    files = []
+    item_files.each do |item_file|
+      files << google_credentials.download_file(item_file.google_file_id, item_file.google_file_name)
+    end
+
+    return unless files.any?
```
[the remark this answers is not in the document]
Comment (lagoan) [reply app/jobs/batch_ingestion_job.rb:133]: This was just meant to stop the rest of the code if there were no files added from the beggining. I agree that it ins't clear.

File: app/jobs/batch_ingestion_job.rb [hunk 3]
```diff
@@ -121,9 +124,15 @@ def item_ingest(batch_ingest, item_data)
     item
   end
 
-  def file_ingest(item, item_file, google_credentials)
-    file = google_credentials.download_file(item_file.google_file_id, item_file.google_file_name)
-    item.add_and_ingest_files([file])
+  def files_ingest(item, item_files, google_credentials)
+    files = []
+    item_files.each do |item_file|
+      files << google_credentials.download_file(item_file.google_file_id, item_file.google_file_name)
```
[the remark this answers is not in the document]
Comment (lagoan) [reply app/jobs/batch_ingestion_job.rb:130]: You are righ, the resulting Tempfiles would still be referenced and stored in memory. I will change it back to ingest each file seperatedly.

[the remark this answers is not in the document]
Comment (lagoan) [reply app/jobs/batch_ingestion_job.rb:130]: It looks like Tempfiles are only stored on the disk. Regardless, with the exception of a couple of tests, all files are ingested individually. I will continue with this behaviour.

**Review by pgwillia — CHANGES_REQUESTED** [review]
I think we're just missing a test of the change to the validator?

https://github.com/ualbertalib/jupiter/blob/09f318b0c6b6996116ebcaa1d33cfc3205468a8f/test/models/batch_ingest_test.rb#L67-L91

Could edit this so that it has two files missing?
https://github.com/ualbertalib/jupiter/blob/09f318b0c6b6996116ebcaa1d33cfc3205468a8f/test/vcr/google_fetch_spreadsheet.yml#L87-L110

i.e. `"conference_logo.png | conference.pdf"`

Then add to the batch_ingest_test.rb from above
```
assert_equal("File(s) 'conference_logo.png, conference.pdf' from row 1 of spreadsheet are not listed in the file list below", 
                     batch_ingest.errors[:google_spreadsheet_id].fourth)
```

**Review by pgwillia — APPROVED** [review]
